### ISSUE TITLE
[Feature Request] The floating action button (FAB) “save changes” when the keyboard is active

### ISSUE THREAD

**Issue by ngocanhtve** [jameskokoska/Cashew#188]
**Is your feature request related to a problem? Please describe.**

I just noticed that every time I edit the notes of a previously entered transaction, the **save changes** button is always at the bottom of the screen and hidden by the keyboard. 

Which leads to one thing, I always have to press the done button on the keyboard to hide it, then reveal the **Save Changes** button and have to press the **Save Changes** button again to complete the editing. 

**Describe the solution you'd like**

Please add a save changes button as a **Floating Action Button** (FAB). Just like the **Add** button on the transactions page, but this time it will only appear in the upper right corner of the keyboard every time the keyboard is activated for an editing operation. Now we can save changes any time we want without having to hide the keyboard anymore.

**Describe alternatives you've considered**

Not thought out! But if possible, send a command to the keyboard so that it temporarily switch the **done** button to the **save** button to directly save the changes!

**Additional context**

A FAB will look something like this:

![](https://i.imgur.com/7Ff7u5I.jpeg)

**Comment by jameskokoska** [comment]
This is a neat idea and I toyed around with it. However, from a usability perspective, having 2 different ways to perform an action (i.e. save changes by hitting the FAB or the bottom save button) leads to a confusing interaction. If Cashew used a FAB for the entirety of the Add Transaction screen, it would make sense, however that isn't the case. Ideally, the [Save Changes] button would be above the entire keyboard, however it blocks a lot of space. 

I do like the idea however, and felt that Cashew needed a better way to minimize the keyboard especially when editing notes and on iOS, so I have made it so a FAB will appear to be able to finish the text box editing for notes. Additionally, I found a bug where the overlay would be rendered twice, so that was a nice fix along the way.

Added here: https://github.com/jameskokoska/Cashew/commit/640258717b9213287bb1f4e08574d92ebd9bfa51
